posts/affine-group-scheme-hopf-algebra.md

```diff
@@ -0,0 +1,257 @@
+  and its group structure is given by
+  [`Diag.grp_ClassAsOver`](https://yaeldillies.github.io/Toric/docs/find/?pattern=AlgebraicGeometry.Scheme.Diag.grp_ClassAsOver#doc).
+
+# Hopf algebras
+
+Since complex affine varieties correspond to commutative $\mathbb{C}$-algebras,
+some $\mathbb{C}$-algebras with extra structure should correspond to algebraic groups.
+
+What are they? Let's figure it out.
+
+Recall that maps $\operatorname{Spec} R \to \operatorname{Spec} S$ of affine varieties
+are in bijection with maps $S \to R$ of their corresponding rings in the opposite direction.
+Also note that we can interpret the product of affine varieties
+in terms of the tensor product of rings:
+$\operatorname{Spec} R \otimes \operatorname{Spec} S$ is isomorphic to $\operatorname{Spec} (R \otimes S)$.
+
+Thus the structure we are looking for is:
+1. A commutative $\mathbb{C}$-algebra $R$.
+2. A **counit homomorphism** $\varepsilon: R\to\mathbb{C}$.
+3. A **comultiplication homomorphism** $\Delta: R \to R \otimes R$.
+4. An **antipode homomorphism** $S: R \to R$.
+
+And we ask that these satisfy the same diagrams as for a group object,
+except that the arrows are reversed:
+<span style="color:red">**TODO: Add diagrams**</span>
+
+Algebras with this structure are called *Hopf algebras*.
+There is a notion of Hopf algebra homomorphisms,
+which are in one-to-one correspondence with algebraic group homomorphisms.
+
+The 1-dimensional torus corresponds to the Hopf algebra defined as:
+1. The $\mathbb{C}$-algebra $\mathbb{C}[t,t^{-1}]$.
+2. The counit $\varepsilon: \mathbb{C}[t,t^{-1}] \to \mathbb{C}$ given by $\varepsilon(t) = 1$.
+3. The comultiplication $\Delta: \mathbb{C}[t,t^{-1}] \to \mathbb{C}[t,t^{-1}] \otimes \mathbb{C}[t,t^{-1}]$
+  given by $\Delta(t) = t \otimes t$.
+4. The antipode $S: \mathbb{C}[t,t^{-1}] \to \mathbb{C}[t,t^{-1}]$ given by $S(t) = t^{-1}$.
+
+> The correspondence between algebraic groups and Hopf algebras is now fully formalized in Toric.
+  We have proven that there is an equivalence of categories
+  between commutative Hopf algebras and affine group schemes.
+  $\Spec$ as a functor from Hopf algebras to affine group schemes is
+  [`hopfSpec`](https://yaeldillies.github.io/Toric/docs/find/?pattern=hopfSpec#doc)
+  and the fact that it's fully faithful is
+  [`hopfSpec.fullyFaithful`](https://yaeldillies.github.io/Toric/docs/find/?pattern=hopfSpec.fullyFaithful#doc).
+
+# Application: Pairing
+
+For a commutative algebraic group $G$, there are two important notions:
+1. A **character** of $G$ is a group homomorphism $G\to\mathbb G_m$.
+  We write $X(G) := \operatorname{Hom}(G, \mathbb G_m)$ for the group of characters.
+2. A **cocharacter**, aka **one-parameter subgroup**, of $G$
+  is a group homomorphism $\mathbb G_m\to G$.
+  We write $X^*(G) := \operatorname{Hom}(\mathbb G_m, G)$ for the group of cocharacters.
+
+Note that characters and cocharacters form genuine honest-to-god groups,
+not merely group objects in some category.
```

Comment:
I think the point is to clarify that other "groups" are not "groups of sets". 
And the last sentence is false in the sense that groups are still "group objects in some category" and is inaccurate in the sense that "group objects in some category" are groups with more structure and are thus a stronger notion than just a group.